kafka/config.go

```diff
@@ -0,0 +1,192 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"log/syslog"
+	"strings"
+	"time"
+
+	kafkalib "github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Supported auth types
+const (
+	AuthTypePlain    = "plain"
+	AuthTypeSCRAM256 = "scram-sha256"
+	AuthTypeSCRAM512 = "scram-sha512"
+)
+
+// DefaultLogLevel is the log level Kafka producers/consumers will use if non set.
+const DefaultLogLevel = logrus.ErrorLevel
+
+// Config holds all the configuration for this package.
+type Config struct {
+	Brokers   []string       `json:"brokers"`
+	Topic     string         `json:"topic"`
+	Producer  ProducerConfig `json:"producer" envconfig:"producer"`
+	Consumer  ConsumerConfig `json:"consumer" envconfig:"consumer"`
```

Comment:
I don't think you need the `envconfig` here, iirc it would mean that you'd have to specify it with _only_ that name and no nesting. 

e.g. for a config like:
```
type Config struct {
  Kafka kafka.Config
}
```
would need to be specified like

```
CONSUMER_GROUP_ID="blah"
```

versus 
```
KAFKA_CONSUMER_GROUP_ID="blah"
```

Please let me know if I'm wrong!

Reply:
Yeah, good point! needed it in the past because the field was called `ProducerConf` and `ConsumerConf`. Gonna change it right now.

Reply:
Done!